src/components/Dialog/ConFirmDeleteDialog.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Alert, Button } from "..";
+import { confirmDialogAction, updateListAction } from "../../store/actions";
+import "../styles/dialog.css";
+import { IoWarningOutline } from "react-icons/io5";
+import {
+  deleteAllRelativeDeviceRequests,
+  deleteDeviceInfoByEmail,
+  deleteOldEmployeeImage,
+} from "../../apiService";
+import toast from "react-hot-toast";
+
+function ConfirmDeleteDialog() {
+  const { shouldDisplay, name, email, imageSrcs } = useSelector((state) => ({
+    shouldDisplay: state.confirmDialog.open,
+    name: state.confirmDialog.name,
+    email: state.confirmDialog.email,
+    imageSrcs: state.confirmDialog.imageSrcs,
+  }));
+  const dispatch = useDispatch();
+
+  const handleClose = () => dispatch(confirmDialogAction.invisible());
+
+  const deleteOldEmployeeDeviceImages = async (images) => {
+    try {
+      if (imageSrcs) {
+        const imageDeletePromises = images.map((imageURL) => {
+          return deleteOldEmployeeImage(imageURL);
+        });
+        await Promise.all(imageDeletePromises);
+      }
+    } catch (error) {
+      //ignore error
+    }
+  };
+
+  const handleDeleteDeviceInfo = async () => {
+    try {
+      await deleteDeviceInfoByEmail(email);
+      await deleteAllRelativeDeviceRequests(email);
+      await deleteOldEmployeeDeviceImages(imageSrcs);
```

Comment:
If they don't depend on each other, can we just call them at the same time instead of calling one by one?